src/storage/patch/SparqlUpdatePatchHandler.ts

```diff
@@ -82,15 +83,23 @@ export class SparqlUpdatePatchHandler extends PatchHandler {
    */
   private async applyPatch(identifier: ResourceIdentifier, deletes: Algebra.Pattern[], inserts: Algebra.Pattern[]):
   Promise<void> {
-    // Read the quads of the current representation
-    const quads = await this.source.getRepresentation(identifier, { type: [{ value: INTERNAL_QUADS, weight: 1 }]});
     const store = new Store<BaseQuad>();
-    const importEmitter = store.import(quads.data);
-    await new Promise((resolve, reject): void => {
-      importEmitter.on('end', resolve);
-      importEmitter.on('error', reject);
-    });
-    this.logger.debug(`${store.size} quads in ${identifier.path}.`);
+    try {
+      // Read the quads of the current representation
+      const quads = await this.source.getRepresentation(identifier, { type: [{ value: INTERNAL_QUADS, weight: 1 }]});
```

Comment:
Yeah, we definitely need #373.

Reply:
Specifically for this case I actually wouldn't use it since we want to read the data anyway if it exists.